pkg/types/types.go

```diff
@@ -428,3 +445,19 @@ type WithNetNsID struct {
 func (e *WithNetNsID) GetNetNSID() uint64 {
 	return e.NetNsID
 }
+
+// parsePodLabels parses the podLabels string into a map
+func parsePodLabels(s string) (map[string]string, error) {
+	labels := map[string]string{}
+	pairs := strings.Split(s, ",")
+	var err error
+	for _, pair := range pairs {
+		kv := strings.SplitN(pair, "=", 2)
+		if len(kv) == 2 {
+			labels[kv[0]] = kv[1]
+		} else {
+			err = fmt.Errorf("invalid pod labels format. Format should be key=value")
```

Comment:
@eiffel-fl - by explode you mean panic? I thought that if we can avoid it then we should.
Regarding the suggested change, if there are several pod labels in the wrong format the error would only show the last invalid pair. We can overcome this by appending all invalid pairs, but possibly that could get quite big error which I thought it's better to avoid.

TBH out of all options I think current implementation is best, but I'll do as you'll decide :)